addons/account/models/account_bank_statement.py

```diff
@@ -523,11 +524,20 @@ class AccountBankStatementLine(models.Model):
              "when the partner doesn't exist yet in the database (or cannot be found).")
     transaction_type = fields.Char(string='Transaction Type')
     payment_ref = fields.Char(string='Label', required=True)
-    amount = fields.Monetary(currency_field='currency_id')
-    amount_currency = fields.Monetary(currency_field='foreign_currency_id',
-        help="The amount expressed in an optional other currency if it is a multi-currency entry.")
-    foreign_currency_id = fields.Many2one('res.currency', string='Foreign Currency',
-        help="The optional other currency if it is a multi-currency entry.")
+    amount = fields.Monetary(
+        string="Amount in Currency",
```

Comment:
That change has been requested by @tsb-odoo .
This can't be a related field since `currency_id` is not required on the Journal. Then, it's `journal.currency_id or journal.company_id.currency_id`

Reply:
Please consider task 2834678 too. currency_id will soon always be set on Journal.

Reply:
> This can't be a related field since `currency_id` is not required on the Journal.

![image](https://user-images.githubusercontent.com/44770049/169866800-5192e521-7876-47f1-9d93-65d6de41d462.png)

Reply:
and set its default to company currency and update them on the write method of company
```python
if 'currency_id' in vals:
    self.env['acoount.journal'].search([('company_id', '=', self.id), ('currency_id', '=', vals['currency_id']).currency_id = vals['currency_id']
```

Reply:
But still it is possible to change the currency after journals are created and before creating any journal entry
```python
@api.depends('company_id.currency_id')
def _compute_currency_id(self):
    for journal in self:
        journal.currency_id = journal.company_id.currency_id
```
We cannot use compute because we need to know the value of the company currency *before* change, because we only need to change the journal currency if it was the same as company's (and keep the foreign currency journals the same)

P.S. We could use `_field_will_change` here if it was defined in base model